tools/sdk-generation-pipeline/pipelines/steps/precheck.yml

```
@@ -0,0 +1,53 @@
+parameters:
+  - name: swagger
+    type: object
+
+  - name: sdk
+    type: string
+    values:
+      - js
+      - java
+      - go
+      - net
+      - python
+
+  - name: service
+    type: string
+steps:
+  - bash: |
+      if [ -z ${{ parameters.swagger.branch }} ];then
+        echo "swagger.branch is empty"
+        exit 1
+      elif [ -z ${{ parameters.swagger.path }} ];then
+        echo "swagger.path is empty"
+        exit 1
+      else
+        echo "##vso[task.setvariable variable=specRepoUrl;]${{ parameters.swagger.path }}"
+        echo "##vso[task.setvariable variable=specRepoBranch;]${{ parameters.swagger.branch }}"
+      fi
+    displayName: Check parameters
+
+  - ${{ if eq(parameters.sdk, 'js') }}:
+    - bash: |
+        echo "##vso[task.setvariable variable=sdkRepoUrl;]https://github.com/Azure/azure-sdk-for-js.git"
```

We can template this similar to:
```suggestion
        echo "##vso[task.setvariable variable=sdkRepoUrl;]https://github.com/Azure/azure-sdk-for-${{parameters.sdk}}.git"
```